src/ThresholdLinear/fixed_point_presheaf.jl

```diff
@@ -1,12 +1,27 @@
+using Catlab.Graphs
+
 # TODO connect to Catlab Sheaf module
 abstract type AbstractPresheaf end
 
 """    The Fixed Point Support Functor is a separated presheaf. This is a section of that presheaf, which associates a neighborhood of a graph with its fixed point supports, if there are any. """
 @struct_hash_equal mutable struct FP <: AbstractPresheaf
     base::Union{Graph, ImplicitGraph}
-    data::Union{FPSections, Missing} 
+    data::Union{FPSections, Missing}
+    hat::Bool
 end
 export FP
+
+# if consumes a ImplicitGraph, then load known information. The docs at this point do not demonstrate "missing" information, as fixed point supports are automatically computed
+function FP(g::Union{Graph, ImplicitGraph}; compute::Bool=true, hat::Bool=true)
+    supports = compute ? FPSections(g) : missing
+    FP(g, supports, hat)
+end
+
+function hat(fp::FP)::FP
+    fp.hat = true
+    fp
```

Comment:
This should construct a copy of the FP rather than mutate, but referring to the same underlying data  If you want to mutate the hat field, then you should rename this to `hat!`